### PR TITLE
Type SettingsContainer's container

### PR DESCRIPTION
**Pull request by akalin-keybase**

### CODE REVIEW

File: shared/settings/index.js [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
-  })
-)(SettingsContainer)
+type StateProps = {
+  badgeNumbers: {[key: Tab]: number},
+  selectedTab: Tab,
+}
+
+const mapStateToProps = (state: TypedState, {routeSelected}: RouteProps<{}, {}>) => ({
+  badgeNumbers: {}, // TODO add badging logic
+  // TODO: Is there a way to validate that routeSelected is a Tab?
```
Comment (akalin-keybase) [inline shared/settings/index.js]: What would be the right way to fix this? Would we have to type `routeSelected` as the union of all possible route types?

File: shared/settings/render.js.flow [hunk 2]
```diff
@@ -7,7 +7,6 @@ export type Props = {
   children: React.Node,
   badgeNumbers: {[key: Tab]: number},
   selectedTab: Tab,
-  showComingSoon: boolean,
```
Comment (akalin-keybase) [inline shared/settings/render.js.flow]: seems unused

File: shared/settings/index.js [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
+import type {Tab} from '../constants/settings'
+import type {TypedState} from '../constants/reducer'
 import type {RouteProps} from '../route-tree/render-route'
 
-// $FlowIssue type this connector
-export default pausableConnect(
-  (state, {routeSelected, routeLeafTags}: RouteProps<{}, {}>) => ({
-    badgeNumbers: {}, // TODO add badging logic
-    selectedTab: routeSelected,
-    isModal: routeLeafTags.modal,
```
Comment (akalin-keybase) [inline shared/settings/index.js]: isModal seems unused

Comment (chrisnojima) [reply shared/settings/index.js]: ??? isn't this using it https://github.com/keybase/client/blob/264a242ee310ba7d5e6faa1a1722225176f9557d/shared/settings/render.desktop.js#L13

File: shared/settings/index.js (hunk 1 above)
Comment (chrisnojima) [reply shared/settings/index.js]: unclear. likely not worth making work now

File: shared/settings/render.js.flow (hunk 2 above)
Comment (chrisnojima) [reply shared/settings/render.js.flow]: should remove it here also: https://github.com/keybase/client/blob/264a242ee310ba7d5e6faa1a1722225176f9557d/shared/settings/dumb.desktop.js#L233

Comment (akalin-keybase) [reply shared/settings/render.js.flow]: Done.

File: shared/settings/index.js (hunk 3 above)
Comment (akalin-keybase) [reply shared/settings/index.js]: Yes, don't know how I missed that. :(

**Comment by akalin-keybase** [comment]
All comments addressed, also removed default `mergeProps` implementations. PTAL!

**Pull request merged**